### DISCUSSION TITLE
DetermineBondOrders doesn't like nitro groups?

### DISCUSSION THREAD

**Discussion by ghutchis** [rdkit/rdkit#6637] [Q&A]
I've been trying the following:
```
        mol = Chem.MolFromXYZFile(xyz_file)
        rdDetermineBonds.DetermineConnectivity(mol)
        rdDetermineBonds.DetermineBondOrders(mol)
```

There are frequent exceptions `Final molecular charge does not match input; could not find valid bond ordering` when the molecule contains a nitro group, e.g.

[HVHNMNGARPCGGD-UHFFFAOYSA-O.xyz.gz](https://github.com/rdkit/rdkit/files/12373054/HVHNMNGARPCGGD-UHFFFAOYSA-O.xyz.gz)

This seems weird to me. If the nitro group is -[N+](O-)=O then there's no change to the molecular charge. If the nitro group is somehow determined as hypervalent, -N(=O)=O it still seems like it should result in no change to the molecular charge.

I'm trying to batch convert a bunch of XYZ files and compare RDKit's new method to Open Babel and a new random forest model. But I don't want to exclude every molecule with nitro groups.

**Comment by heseng0** [comment]
I faced the same problem. In some cases, a workaround can be to add -2 to the total charge for -NO2. This results in a trivalent N and both oxygens negatively charged (which is not ideal, but working fine for my usage). Maybe this also also works in your case

**Comment by smcolby** [comment]
Also seeing this issue with nitro groups. I am trying to simulate vapor phase infrared, which is for "neutral" molecules. The positive and negative partial charges of the nitrogen and single-bonded oxygen should keep the molecule neutral, but the logic in determining bond connectivity seems to not support such cases.

This issue appears with version 2023.09.5, but not 2022.09.4.

**Comment by JEAM-professional** [comment]
I have been facing similar issues with NO2 and a few others (thiophene substitutions in aromatic rings, if i recall correctly). Has there, by chance, been an update here?

**Comment by dehaenw** [comment]
In the given example "HVHNMNGARPCGGD-UHFFFAOYSA-O.xyz" the fix is simple. The total charge of this species is actually +1", if you inspect the coords you'll see one of the aniline nitrogens is charged. So you just specify the charge with `rdDetermineBonds.DetermineBondOrders(mol,charge=1)` it works
